py_zipkin/encoding/_decoders.py

```diff
@@ -102,20 +112,29 @@ def _convert_from_thrift_endpoint(self, thrift_endpoint):
             port=port,
         )
 
-    def _decode_thrift_annotations(self, thrift_annotations):
+    def _decode_thrift_annotations(
+        self, thrift_annotations: List[zipkinCore.Annotation]
+    ) -> Tuple[
+        Dict[str, Optional[float]],
+        Optional[Endpoint],
+        Kind,
+        Optional[int],
+        Optional[int],
+    ]:
         """Accepts a thrift annotation and converts it to a v1 annotation.
 
         :param thrift_annotations: list of thrift annotations.
-        :type thrift_annotations: list of zipkin_core.Span.Annotation
-        :returns: (annotations, local_endpoint, kind)
+        :type thrift_annotations: list of zipkinCore.Span.Annotation
+        :returns: (annotations, local_endpoint, kind, timestmap, duration)
         """
         local_endpoint = None
         kind = Kind.LOCAL
-        all_annotations = {}
-        timestamp = None
-        duration = None
+        all_annotations: Dict[str, Optional[int]] = {}
+        timestamp: Optional[int] = None
```

Comment:
Optional[float] ?

Reply:
Timestamps are ints of microseconds in this context.